contracts/src/periphery/ControllerV1.sol

```diff
@@ -101,36 +101,83 @@ contract ControllerV1 is Ownable {
         uint24 bundleFee,
         uint24 unlockedFee
     ) external {
-        _checkOwner();
+        _checkOwnerOrNode();
+
+        // Call to `.configurePool` will check for us whether `asset0 == asset1`.
         if (asset0 > asset1) (asset0, asset1) = (asset1, asset0);
-        StoreKey key = PoolConfigStoreLib.keyFromAssetsUnchecked(asset0, asset1);
-        pools[key] = Pool(asset0, asset1);
+        StoreKey key = StoreKeyLib.keyFromAssetsUnchecked(asset0, asset1);
+
+        uint256 maybe_index = _poolIndices[key];
+        if (maybe_index == 0) {
+            _pools.push(Pool(asset0, asset1));
+            maybe_index = _pools.length;
+            _poolIndices[key] = maybe_index;
+        }
 
         emit PoolConfigured(asset0, asset1, tickSpacing, bundleFee, unlockedFee);
         ANGSTROM.configurePool(asset0, asset1, tickSpacing, bundleFee, unlockedFee);
     }
 
     function removePool(address asset0, address asset1) external {
-        _checkOwner();
+        _checkOwnerOrNode();
```

Comment:
We want to keep it so that adding and removing pools are strictly out of the nodes control and controlled by the DAO instead. Will be easier on the backend as then we can keep this logic out of the node for adding and removing pools and it can just index instead. The only thing the node should be able to modify is the fee